### DISCUSSION TITLE
First tests with 3d mapping with lasers

### DISCUSSION THREAD

**Discussion by bhowiebkr** [bhowiebkr/laser-level-webcam#19] [Announcements]
See the video for how it works https://youtu.be/wpInsrcQAus?si=OUiFMbnuRa_l9gGx

![IMG_20230915_124509](https://github.com/bhowiebkr/laser-level-webcam/assets/904170/007ff9b4-aa43-44a0-874e-4f832c1d4472)

I've also got the 3d surface graph hooked up now, some other things like saving the state of the GUI, etc.

I'm working on some specific gcode programs now just probing the surface and after that probing and machining the offset flat.

**Comment by bhowiebkr** [comment]
Working with Python sockets has been a real pain managing them. Stuff like closing them, reconnecting, etc, does not properly work . Until I figure that out, if someone else wants to have a try at it and do a pull request that’d be great. For now you'll have to shutdown both the server and client tools and boot them up again to get a socket connection between them working.

**Comment by miked243** [comment]
This is pretty cool, i watched the video of you running it on your machine. I wonder how difficult it would be to write a program similar to this for mapping a granite surface plate. Just using the hight gauge by hand like you would an autocollimator or something similar.
